src/plugins/workflow_designer/src/cmdline/WorkflowCMDLineTasks.cpp

```diff
@@ -55,44 +55,44 @@ WorkflowRunFromCMDLineBase::WorkflowRunFromCMDLineBase()
       workflowRunTask(NULL) {
     GCOUNTER(cvar, "workflow_run_from_cmdline");
 
-    CMDLineRegistry *cmdLineRegistry = AppContext::getCMDLineRegistry();
-
-    // try to process schema without 'task' option (it can only be the first one)
-    QStringList pureValues = CMDLineRegistryUtils::getPureValues();
-    if (!pureValues.isEmpty()) {
-        QString schemaName = pureValues.first();
-        processLoadSchemaTask(schemaName, 1);    // because after program name
-    }
-    if (loadTask != NULL) {
-        addSubTask(loadTask);
-        return;
-    }
-
-    // process schema with 'task' option
-    int taskOptionIdx = CMDLineRegistryUtils::getParameterIndex(WorkflowDesignerPlugin::RUN_WORKFLOW);
-    if (taskOptionIdx != -1) {
-        processLoadSchemaTask(cmdLineRegistry->getParameterValue(WorkflowDesignerPlugin::RUN_WORKFLOW, taskOptionIdx), taskOptionIdx);
-    }
-    if (loadTask == NULL) {
-        setError(tr("no task to run"));
-        return;
-    }
-    addSubTask(loadTask);
+    connect(this, &Task::si_stateChanged, this, &WorkflowRunFromCMDLineBase::sl_stateChanged);
 }
 
 void WorkflowRunFromCMDLineBase::processLoadSchemaTask(const QString &schemaNameCandidate, int optionIdx) {
     loadTask = prepareLoadSchemaTask(schemaNameCandidate);
+    CHECK_OP(stateInfo, );
+
     if (loadTask != NULL) {
         schemaName = schemaNameCandidate;
         optionsStartAt = optionIdx + 1;
     }
 }
 
+void WorkflowRunFromCMDLineBase::sl_stateChanged() {
+    CHECK(isFinished(), );
+    CHECK(hasError(), );
+
+    CMDLineRegistry* cmdLineRegistry = AppContext::getCMDLineRegistry();
+    SAFE_POINT(cmdLineRegistry != nullptr, "CMDLineRegistry is nullptr", );
+
+    const QString errorStateFilePath = cmdLineRegistry->getParameterValue(CmdlineTaskRunner::ERROR_STATE_FILE_ARG);
```

Comment:
nit: const for locals: some locals with no const some with const - inconsistent style within a single PR. I would suggest skipping const for locals to be consisitent with the rest UGENE code.

Reply:
OK